pyuploadcare/api_resources.py

```diff
@@ -231,25 +231,45 @@ def store(self):
         """
         self._info_cache = rest_request('PUT', self._api_storage_uri)
 
-    def copy(self, effects=None, target=None):
+    def copy(self, **kwargs):
         """Creates File copy
-
-        If ``target`` is ``None``, copy file to Uploadcare storage otherwise
-        copy to target associated with project.
-        Add ``effects`` to ``self.default_effects`` if any.
-        """
-        effects = effects or ''
+        Keyword Args:
+          - target:  
+                If ``target`` is ``None``, copy file to Uploadcare storage otherwise
+                copy to target associated with project;
+          - effects: 
+                Add ``effects`` to ``self.default_effects`` if any;
+          - make_public: 
+                To forbid a public from accessing your files on the target storage set
+                ``make_public`` option to be False, relevant when ``target`` has a value set.
+          - store: 
+                If ``store`` option is set to False the copy of your file will be deleted
+                in 24 hour period after the upload.
+          - pattern: 
+                Specify  ``pattern`` option to set S3 object key name.
+
+        Following example copies a file to ``samplefs`` S3 storage with  owner only access, 
+        and file name consisting of uuid concatenated to original filename  placed
+        in the root of the bucket:
+             >>>file.copy(target='samplefs',
+                                make_public=False,
+                                pattern='${uuid}${filename}${ext}');
+
+        In this example, a file is copied to ``samplefs`` storage with public access and placed
+        under the directory named after file uuid. A new file has the same file name as its original.
+                >>>file.copy(None,'samplefs');
+
```

Comment:
you can borrow descriptions directly from https://uploadcare.com/documentation/rest/#file